src/antelope/wallets/ual/MetakeepUAL.ts

```diff
@@ -0,0 +1,534 @@
+     */
+    getError(): UALError | null {
+        return null;
+    }
+
+    /**
+     * Returns true if the authenticator is loading while initializing its internal state.
+     */
+    isLoading() {
+        return this.loading;
+    }
+
+    /**
+     * Returns the style of the Button that will be rendered.
+     */
+    getStyle() {
+        return {
+            // An icon displayed to app users when selecting their authentication method
+            icon: 'no-icon',
+            // Name displayed to app users
+            text: metakeep_name,
+            // Background color displayed to app users who select your authenticator
+            background: '#030238',
+            // Color of text used on top the `backgound` property above
+            textColor: '#FFFFFF',
+        };
+    }
+
+    /**
+     * Returns whether or not the button should render based on the operating environment and other factors.
+     * ie. If your Authenticator App does not support mobile, it returns false when running in a mobile browser.
+     */
+    shouldRender() {
+        return true;
+    }
+
+    /**
+     * Returns whether or not the dapp should attempt to auto login with the Authenticator app.
+     * Auto login will only occur when there is only one Authenticator that returns shouldRender() true and
+     * shouldAutoLogin() true.
+     */
+    shouldAutoLogin() {
+        return true;
+    }
+
+    /**
+     * Returns whether or not the button should show an account name input field.
+     * This is for Authenticators that do not have a concept of account names.
+     */
+    async shouldRequestAccountName() {
+        return false;
+    }
+
+    async createAccount(publicKey: string): Promise<string> {
+        const suggestedName = await this.accountNameSelector.selectAccountName();
+        return axios.post(this.accountCreateAPI, {
+            ownerKey: publicKey,
+            activeKey: publicKey,
+            jwt: this.userCredentials.jwt,
+            suggestedName: suggestedName,
+        }).then(response => response.data.accountName);
+    }
+
+    resolveAccountName() {
+        return new Promise<string>(async (resolve, reject) => {
+            let accountName = '';
+            if (!metakeep) {
+                return reject(new Error('metakeep is not initialized'));
+            }
+            if (this.userCredentials.email === '') {
+                return reject(new Error('No account email'));
+            }
+
+            // we check if we have the account name in the cache
+            const accountNames = metakeepCache.getAccountNames(this.userCredentials.email, this.chainId);
+            if (accountNames.length > 0) {
+                if (accountNames.length > 1) {
+                    // if we have more than one account, we ask the user to select one using this callback
+                    const selectedAccount = await this.accountSelector.selectAccount(accountNames);
```

Comment:
```ts
const selectedAccount = await this.accountSelector.selectAccount(accountNames);
```
That line is a callback to the outside website to stop the flow and allow the user to choose from of the found accounts to log in with. This is optional and if not overwritten, a default simple selection with no interaction will be executed instead.